server/jetstream_api.go

```diff
@@ -1699,10 +1707,47 @@ func (s *Server) jsLeaderStepDownRequest(sub *subscription, c *client, subject,
 		return
 	}
 
+	var preferredLeader string
+	var resp = JSApiLeaderStepDownResponse{ApiResponse: ApiResponse{Type: JSApiLeaderStepDownResponseType}}
+
+	if !isEmptyRequest(msg) {
+		var req JSApiLeaderStepdownRequest
+		if err := json.Unmarshal(msg, &req); err != nil {
+			resp.Error = jsInvalidJSONErr
+			s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
+			return
+		}
+		if len(req.Placement.Tags) > 0 {
+			// Tags currently not supported.
+			resp.Error = jsClusterTagsErr
+			s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
+			return
+		}
+		cn := req.Placement.Cluster
+		var peers []string
+		for _, p := range cc.meta.Peers() {
+			si, ok := s.nodeToInfo.Load(p.ID)
+			ni := si.(*nodeInfo)
```

Comment:
if `!ok` then `si` would be nil, no? It may be a case where you need 2 separate `if`'s.

Reply:
Good point, will fix.

Reply:
ok take another look.